features/step_definitions/provider_steps.rb

```diff
@@ -247,9 +247,11 @@ def create_provider_with_plan(name, plan) # TODO: RENAME THIS NOWWW
   @provider = FactoryBot.create(:provider_account_with_pending_users_signed_up_to_no_plan, org_name: name,
                                                                                            domain: name,
                                                                                            self_domain: "admin.#{name}")
-  @provider.application_contracts.delete_all
-
-  @provider.buy!(plan, name: 'Default', description: 'Default') unless @provider.bought?(plan)
+  if plan && !@provider.bought?(plan)
```

Comment:
All is fine, I'm just wondering, we created a provider `provider_account_with_pending_users_signed_up_to_no_plan`, so it is supposed not to have bought a plan. And also why allow method to be called without a `plan`? 

But anyway, should work like this.

Reply:
Uhm, actually, it's indeed not needed to verify the `plan`, because it's always used with a plan. But, following the "call to action" in the comment, I tried to refactor this method, and call it `def create_provider(name:, plan: nil)`

Because well, the new name kind of doesn't require a plan. But then I reverted the change, but this is left.

I can remove the `if plan` and keep only `unless @provider.bought?(plan)` if you think it's better.

But the actual change (which not sure if correct or not), is that deleting all contracts is now under the condition, while it used to be "outside". I just thought it did not make sense to delete all applications if were not going to create a new one.

Reply:
Things should be ok now, I'm fine whatever you decide.